### PR TITLE
Match exactly analytics

### LINKED ISSUE #3958: Add analytics for Match Exactly
~~Blocked by #3959 / #3940.~~

* Send event with server hash when clicking refresh button
* Set user property for selected "form update" mode ("manually", "match_exactly" and "previously_downloaded_only")
* Send event when a sync completes with failure or success (like storage migration). Include error type with failures.

### PR DESCRIPTION
**Pull request by seadowg**
Closes #3958

#### What has been done to verify that this works as intended?

New tests.

#### Why is this the best possible solution? Were any other approaches considered?

Comments inline

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Just adds analytics so should be good to merge as long as we try it out properly with the Play Store creds in the app.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.java
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.java:83]: I wanted to do this here rather than when the settings are changed/imported as it avoids handling all the different possible cases and also makes sure we always have a value for this property set. It means the values we have are probably going to be slightly behind (as we'll be waiting for the next time the app boots for the property to be set) but I think that'll be ok.

File: collect_app/src/main/java/org/odk/collect/android/formmanagement/BlankFormsListViewModel.java
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/formmanagement/BlankFormsListViewModel.java]: What do you think of using the hostname instead of the raw setting value? The raw server setting value would mean we differentiate between different Central App Users. Hostname would give us a sense of organizations that use match exactly and specifically the manual sync button.

Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/formmanagement/BlankFormsListViewModel.java]: Yeah that makes more sense

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
@getodk/testers are currently doing their work off master so I think it's best to get the green light from them before merging. This doesn't affect behavior but it does move code and tests around.

**Comment by lognaturel** [comment]
Merging because it will be good to have for the beta.

**Pull request merged**
